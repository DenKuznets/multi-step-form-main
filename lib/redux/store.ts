import { configureStore } from '@reduxjs/toolkit';
import appReducer from './slices/appSlice';
// ...
/* eslint-disable no-underscore-dangle */

export const store = configureStore({
    reducer: {
        app: appReducer
    }
});
/* eslint-enable */
// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
