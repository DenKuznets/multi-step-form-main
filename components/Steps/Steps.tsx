'use client';
import { useAppSelector } from '@/lib/redux/hooks';
import { selectCurrentStep, selectSteps } from '@/lib/redux/slices/appSlice';
import { PropsWithChildren } from 'react';

const Steps = (props: PropsWithChildren) => {
    const currentStep = useAppSelector(selectCurrentStep);
    const steps = useAppSelector(selectSteps);
    const stepsList = steps.map((step, index) => {
        const stepNumber = index + 1;
        return (
            <li className={`items-center sm:flex sm:gap-x-4`} key={step.name}>
                <div
                    className={`${
                        stepNumber === currentStep
                            ? 'bg-lightBlue text-marineBlue'
                            : 'bg-transparent text-white'
                    } flex h-8 w-8 cursor-pointer items-center justify-center rounded-full border text-center font-bold `}
                >
                    {stepNumber}
                </div>
                <div className="hidden flex-col justify-center gap-y-1 sm:flex">
                    <div className="text-lightGray text-[0.8rem] uppercase leading-none ">
                        Step {stepNumber}
                    </div>
                    <div className="font-bold leading-none text-white">
                        {step.name}
                    </div>
                </div>
            </li>
        );
    });
    return (
        <div className="relative flex min-h-screen flex-[1] flex-col items-center rounded-xl sm:m-auto sm:h-[37.5rem] sm:min-h-[unset] sm:max-w-[59rem] sm:flex-row sm:items-start sm:bg-white sm:p-4">
            <ul className="flex h-44 w-full flex-row justify-center gap-x-[1.1rem] bg-[url('/images/bg-sidebar-mobile.svg')] bg-[length:100%_100%] bg-no-repeat pt-8 sm:h-full sm:max-w-[17.4rem] sm:flex-col sm:justify-normal sm:gap-y-8 sm:bg-[url('/images/bg-sidebar-desktop.svg')] sm:pl-8 sm:pt-10">
                {stepsList}
            </ul>
            <div className="relative top-[-5rem] px-4 sm:static sm:h-full sm:w-full sm:p-0">
                {props.children}
            </div>
        </div>
    );
};

export default Steps;
