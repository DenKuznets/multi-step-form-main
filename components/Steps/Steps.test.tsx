import '@testing-library/jest-dom';
import Steps from './Steps';
import { renderWithProviders, screen } from '@/lib/redux/utils-for-tests';

test('renders correctly', () => {
    renderWithProviders(<Steps />);
    const steps = screen.getAllByRole('listitem');
    expect(steps).toHaveLength(4);
});

test('renders Home unchanged', () => {
    const { container } = renderWithProviders(<Steps />);
    expect(container).toMatchSnapshot();
});