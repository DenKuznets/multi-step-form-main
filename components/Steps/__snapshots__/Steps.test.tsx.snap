// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Home unchanged 1`] = `
<div>
  <div
    class="relative flex min-h-screen flex-[1] flex-col items-center rounded-xl sm:m-auto sm:h-[37.5rem] sm:min-h-[unset] sm:max-w-[59rem] sm:flex-row sm:items-start sm:bg-white sm:p-4"
  >
    <ul
      class="flex h-44 w-full flex-row justify-center gap-x-[1.1rem] bg-[url('/images/bg-sidebar-mobile.svg')] bg-[length:100%_100%] bg-no-repeat pt-8 sm:h-full sm:max-w-[17.4rem] sm:flex-col sm:justify-normal sm:gap-y-8 sm:bg-[url('/images/bg-sidebar-desktop.svg')] sm:pl-8 sm:pt-10"
    >
      <li
        class="items-center sm:flex sm:gap-x-4"
      >
        <div
          class="bg-lightBlue text-marineBlue flex h-8 w-8 cursor-pointer items-center justify-center rounded-full border text-center font-bold transition-all"
        >
          1
        </div>
        <div
          class="hidden flex-col justify-center gap-y-1 sm:flex"
        >
          <div
            class="text-lightGray text-[0.8rem] uppercase leading-none "
          >
            Step 
            1
          </div>
          <div
            class="font-bold leading-none text-white uppercase"
          >
            personal info
          </div>
        </div>
      </li>
      <li
        class="items-center sm:flex sm:gap-x-4"
      >
        <div
          class="bg-transparent text-white flex h-8 w-8 cursor-pointer items-center justify-center rounded-full border text-center font-bold transition-all"
        >
          2
        </div>
        <div
          class="hidden flex-col justify-center gap-y-1 sm:flex"
        >
          <div
            class="text-lightGray text-[0.8rem] uppercase leading-none "
          >
            Step 
            2
          </div>
          <div
            class="font-bold leading-none text-white uppercase"
          >
            SELECT PLAN
          </div>
        </div>
      </li>
      <li
        class="items-center sm:flex sm:gap-x-4"
      >
        <div
          class="bg-transparent text-white flex h-8 w-8 cursor-pointer items-center justify-center rounded-full border text-center font-bold transition-all"
        >
          3
        </div>
        <div
          class="hidden flex-col justify-center gap-y-1 sm:flex"
        >
          <div
            class="text-lightGray text-[0.8rem] uppercase leading-none "
          >
            Step 
            3
          </div>
          <div
            class="font-bold leading-none text-white uppercase"
          >
            ADD-ONS
          </div>
        </div>
      </li>
      <li
        class="items-center sm:flex sm:gap-x-4"
      >
        <div
          class="bg-transparent text-white flex h-8 w-8 cursor-pointer items-center justify-center rounded-full border text-center font-bold transition-all"
        >
          4
        </div>
        <div
          class="hidden flex-col justify-center gap-y-1 sm:flex"
        >
          <div
            class="text-lightGray text-[0.8rem] uppercase leading-none "
          >
            Step 
            4
          </div>
          <div
            class="font-bold leading-none text-white uppercase"
          >
            SUMMARY
          </div>
        </div>
      </li>
    </ul>
    <div
      class="relative top-[-5rem] px-4 sm:static sm:h-full sm:w-full sm:p-0"
    />
  </div>
</div>
`;

exports[`renders Steps unchanged 1`] = `
<div>
  <div
    class="relative flex min-h-screen flex-[1] flex-col items-center rounded-xl sm:m-auto sm:h-[37.5rem] sm:min-h-[unset] sm:max-w-[59rem] sm:flex-row sm:items-start sm:bg-white sm:p-4"
  >
    <ul
      class="flex h-44 w-full flex-row justify-center gap-x-[1.1rem] bg-[url('/images/bg-sidebar-mobile.svg')] bg-[length:100%_100%] bg-no-repeat pt-8 sm:h-full sm:max-w-[17.4rem] sm:flex-col sm:justify-normal sm:gap-y-8 sm:bg-[url('/images/bg-sidebar-desktop.svg')] sm:pl-8 sm:pt-10"
    >
      <li
        class="items-center sm:flex sm:gap-x-4"
      >
        <div
          class="bg-lightBlue text-marineBlue flex h-8 w-8 cursor-pointer items-center justify-center rounded-full border text-center font-bold transition-all"
        >
          1
        </div>
        <div
          class="hidden flex-col justify-center gap-y-1 sm:flex"
        >
          <div
            class="text-[0.8rem] uppercase leading-none text-lightGray "
          >
            Step 
            1
          </div>
          <div
            class="font-bold uppercase leading-none text-white"
          >
            personal info
          </div>
        </div>
      </li>
      <li
        class="items-center sm:flex sm:gap-x-4"
      >
        <div
          class="bg-transparent text-white flex h-8 w-8 cursor-pointer items-center justify-center rounded-full border text-center font-bold transition-all"
        >
          2
        </div>
        <div
          class="hidden flex-col justify-center gap-y-1 sm:flex"
        >
          <div
            class="text-[0.8rem] uppercase leading-none text-lightGray "
          >
            Step 
            2
          </div>
          <div
            class="font-bold uppercase leading-none text-white"
          >
            select plan
          </div>
        </div>
      </li>
      <li
        class="items-center sm:flex sm:gap-x-4"
      >
        <div
          class="bg-transparent text-white flex h-8 w-8 cursor-pointer items-center justify-center rounded-full border text-center font-bold transition-all"
        >
          3
        </div>
        <div
          class="hidden flex-col justify-center gap-y-1 sm:flex"
        >
          <div
            class="text-[0.8rem] uppercase leading-none text-lightGray "
          >
            Step 
            3
          </div>
          <div
            class="font-bold uppercase leading-none text-white"
          >
            add-ons
          </div>
        </div>
      </li>
      <li
        class="items-center sm:flex sm:gap-x-4"
      >
        <div
          class="bg-transparent text-white flex h-8 w-8 cursor-pointer items-center justify-center rounded-full border text-center font-bold transition-all"
        >
          4
        </div>
        <div
          class="hidden flex-col justify-center gap-y-1 sm:flex"
        >
          <div
            class="text-[0.8rem] uppercase leading-none text-lightGray "
          >
            Step 
            4
          </div>
          <div
            class="font-bold uppercase leading-none text-white"
          >
            summary
          </div>
        </div>
      </li>
    </ul>
    <div
      class="relative top-[-5rem] px-4 sm:static sm:h-full sm:w-full sm:p-0"
    >
      <div
        class="step sm:relative sm:h-full"
      >
        <h1
          class="mb-2 text-[1.55rem] capitalize sm:mb-[0.9rem] sm:text-[2.05rem]"
        >
          personal info
        </h1>
        <p
          class="text-coolGray mb-4 text-[1.03rem] font-medium tracking-[-.01em] sm:mb-8 sm:tracking-tight"
        >
          Please provide your name, email address, and phone number.
        </p>
      </div>
    </div>
  </div>
</div>
`;
